packages/flutter_blue/tizen/src/bluetooth_manager.cc

```diff
@@ -0,0 +1,479 @@
+#include "bluetooth_manager.h"
+
+#include <bluetooth.h>
+#include <system_info.h>
+#include <tizen.h>
+
+#include <mutex>
+
+#include "GATT/bluetooth_characteristic.h"
+#include "GATT/bluetooth_descriptor.h"
+#include "GATT/bluetooth_service.h"
+#include "bluetooth_device_controller.h"
+#include "log.h"
+
+namespace flutter_blue_tizen {
+
+using State = BluetoothDeviceController::State;
+
+BluetoothManager::BluetoothManager(NotificationsHandler& notificationsHandler)
+    : notifications_handler_(notificationsHandler) {
+  int ret = IsBLEAvailable();
+  if (ret == 0) {
+    LOG_ERROR("Bluetooth is not available on this device!",
+              get_error_message(ret));
+    return;
+  }
+  ret = bt_initialize();
+  if (ret != 0) {
+    LOG_ERROR("[bt_initialize]", get_error_message(ret));
+    return;
+  }
+  ret = bt_gatt_set_connection_state_changed_cb(
+      &BluetoothDeviceController::ConnectionStateCallback, this);
```

Comment:
Directly calling `bt_gatt_set_connection_state_changed_cb` here forces the class to hold `NotificationsHandler` and depend on protobuf data types. How about:
- https://github.com/HakkyuKim/plugins/commit/fa940cde3f1331b881bfc4c5be81fac4a354a197
- https://github.com/HakkyuKim/plugins/commit/460f85b9a2ce81af8e94b071feca86b32852d615